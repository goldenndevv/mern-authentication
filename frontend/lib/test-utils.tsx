import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { theme, Fonts } from "../infrastructure/theme";

const customTheme = extendTheme({
  colors: theme.colors,
  fonts: theme.fonts,
});

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <Fonts />
      {children}
    </ChakraProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "queries">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
