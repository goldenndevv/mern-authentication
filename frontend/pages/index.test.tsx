import { render } from "test-utils"
import { ChakraProvider } from "@chakra-ui/react";

import IndexPage from ".";

describe("Index page", () => {
  it("should match the snapshot", () => {
    const launch = {
      timestamp: 1605401340000,
      mission: "Mission Name",
      site: "Kennedy Space Center",
      rocket: "Falcon 9",
    };
    const { container } = render(<IndexPage launch={launch} />)
    expect(container.firstChild).toMatchSnapshot();
  });
});
