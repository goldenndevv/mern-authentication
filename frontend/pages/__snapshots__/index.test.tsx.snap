// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match the snapshot 1`] = `
<main>
  <h2
    class="chakra-heading css-8iifz"
  >
    Next SpaceX Launch: 
    Mission Name
  </h2>
  <div
    class="css-16lufbi"
  >
    <p
      class="chakra-text css-1u8qly9"
    >
      Falcon 9
       will take off from 
      Kennedy Space Center
       on 
      Sun Nov 15 2020
    </p>
  </div>
</main>
`;
